### ISSUE TITLE
Application crashed when incoming notification received

### ISSUE THREAD

**Issue by qoqobolo** [status-im/status-desktop#8191]
# Bug Report

## Description

The desktop application started crashing immediately after logging in for the main user that I've been using for several days. So now I'm blocked from accessing this account.

I've updated my macOS to the latest version - Ventura 13.0, since it helped other users (e.g. [this issue](https://github.com/status-im/status-desktop/issues/8161)), but it didn't work for me.

Looks like the app crashes after loading the db (see video below). But the exact reason is still unclear. What I could recall:

1. I have two users, both have a community (Status community on the main account and test one on the second)
2. Touch ID is enabled for both users. I tried to log in with both touch ID and password - unsuccessfully
3. But I still can log in to my second test account with one contact, one 1-1 chat, and one community
4. On the main account I have the Status community, 74 contacts, 14 1-1 chats
5. The main account is also restored on the mobile app
6. During the day, I switched between users many times
7. Today I updated the Desktop app to the latest develop 0.7.9, but I was able to use it for a while after the update
8. Both users had the issue with receiving messages

Video  https://drive.google.com/file/d/1ZM7zLbq6HjiWTdN6_TrePcWcrFcGX3uq/view?usp=sharing

Logs from terminal: [desktop_crash1.log](https://github.com/status-im/status-desktop/files/9973076/desktop_crash1.log)

#### Chronology of events

1. Installed Status 3/11
2. Restored a multiaccount from the same seed phrase on Desktop and Mobile
3. Used it for one week
4. Created another multiaccount 
5. Updated version from 0.7.0 to 0.7.9
6. Used it for a while (a couple of hours)

#### Expected behavior

Can log in to both users

#### Actual behavior

Can only log in to the additional test user

### Additional Information

- Status desktop version: 0.7.9
- Operating System: MacBook Pro, macOS Ventura 13.0

**Comment by caybro** [comment]
Ok, this is an actual crash; crashes on an incoming notification:

```
/Users/jenkins/workspace/status-desktop/platforms/macos/src/app_service/service/activity_center/service.nim(178) asyncActivityNotificationLoaded
/Users/jenkins/workspace/status-desktop/platforms/macos/src/app_service/service/activity_center/dto/notification.nim(94) parseActivityCenterNotifications
/Users/jenkins/workspace/status-desktop/platforms/macos/src/app_service/service/activity_center/dto/notification.nim(76) toActivityCenterNotificationDto
/Users/jenkins/workspace/status-desktop/platforms/macos/vendor/nimbus-build-system/vendor/Nim/lib/system/fatal.nim(49) sysFatal
Error: unhandled exception: value out of range: 10 notin 0 .. 9 [RangeError]
```

**Comment by caybro** [comment]
cc @MishkaRogachev ^^

**Comment by MishkaRogachev** [comment]
> Ok, this is an actual crash; crashes on an incoming notification:
> 
> ```
> /Users/jenkins/workspace/status-desktop/platforms/macos/src/app_service/service/activity_center/service.nim(178) asyncActivityNotificationLoaded
> /Users/jenkins/workspace/status-desktop/platforms/macos/src/app_service/service/activity_center/dto/notification.nim(94) parseActivityCenterNotifications
> /Users/jenkins/workspace/status-desktop/platforms/macos/src/app_service/service/activity_center/dto/notification.nim(76) toActivityCenterNotificationDto
> /Users/jenkins/workspace/status-desktop/platforms/macos/vendor/nimbus-build-system/vendor/Nim/lib/system/fatal.nim(49) sysFatal
> Error: unhandled exception: value out of range: 10 notin 0 .. 9 [RangeError]
> ```

It seems that status-go version not compatible with status-desktop. I got this error when accidentally built **0.7.x** with status-go **develop** branch. If 0.7.9 assembled with status-go develop, it should not work at all, we need to add some guards to stop building incompatible versions e.g having same release branch naming in all status repos

**Comment by qoqobolo** [comment]
Hi @caybro @MishkaRogachev !
Could you tell me is there are any updates on this?

I'm still blocked from using Status Desktop, so I was wondering if you guys mind if I reinstall the app today? I'm keeping this for now in case it helps fix it, so let me know if I shouldn't delete the crashed version.
I'm using the mobile version all this time but would like to get back to using Desktop :)

**Comment by MishkaRogachev** [comment]
> Hi @caybro @MishkaRogachev ! Could you tell me is there are any updates on this?
> 
> I'm still blocked from using Status Desktop, so I was wondering if you guys mind if I reinstall the app today? I'm keeping this for now in case it helps fix it, so let me know if I shouldn't delete the crashed version. I'm using the mobile version all this time but would like to get back to using Desktop :)

That fixed by @PascalPrecht in master

**Comment by PascalPrecht** [comment]
@qoqobolo this has been address here: https://github.com/status-im/status-desktop/commit/d9c04bb32dae0e85242b586cca93c98407e5d813

At least for cases where contact verification requests are sent (which I'm not sure you did).
But judging by the actual stack traces posted here, it seems very much related to the fixes I've done.

**Comment by qoqobolo** [comment]
> At least for cases where contact verification requests are sent (which I'm not sure you did).

I had two pending **responses** to verification requests (my contacts sent me requests with questions and I answered them, but they had the `Pending` status), sorry I forgot to mention this in the issue description. Maybe the app might have started crashing since I got approval for these requests? I'm not sure what is the flow exactly though, will figure it out.
Also, I'll test this scenario to make sure that this case works fine as well.

Thanks for the fix @PascalPrecht!
And just to make sure, is it included in release 0.8.0? 
Can I use [this build](https://github.com/status-im/status-desktop/releases/tag/0.8.0-rc.1)?

**Comment by PascalPrecht** [comment]
> Maybe the app might have started crashing since I got approval for these requests? I'm not sure what is the flow exactly though, will figure it out.

@qoqobolo Yes, if you did indeed respond to the verification request then this is exactly what will cause a crash eventually, without the fix I've linked.

> Can I use [this build](https://github.com/status-im/status-desktop/releases/tag/0.8.0-rc.1)?

Oh cool, didn't know we put it out already. But that release will have the fix, yes!

**Comment by PascalPrecht** [comment]
Also @qoqobolo once you've tested this with the build and verified that this isn't an issue anymore, it'd be wonderful if you could close this issue as well :)

**Comment by qoqobolo** [comment]
@PascalPrecht of course, was going to do this :)

> if you did indeed respond to the verification request then this is exactly what will cause a crash eventually

Tested sending a response - works fine for me in terms of normal re-login into the application, with no crashes.

However, I've also faced this issue https://github.com/status-im/status-desktop/issues/8277.
And the two old verification requests are still showing up as `Pending` in my account even though my identity was confirmed by at least one of them (using one of the previous versions of the app). 

So yeah, it works for me now, I can use the desktop app again 🎉
Closing the issue.
